File: ComPDFKit 1.11.0 for Windows/Examples/Samples/AnnotationImportExportTest/VB/AnnotationImportExportTest.vb
Imports ComPDFKit.PDFDocument
Imports System
Imports System.IO

Module AnnotationImportExportTest
    Private parentPath = Path.GetDirectoryName(Path.GetDirectoryName(Path.GetDirectoryName(System.IO.Directory.GetCurrentDirectory())))
    Private outputPath As String = Path.Combine(parentPath, "Output", "VB")
    Private tempPath As String = Path.Combine(outputPath, "temp")

    Sub Main()
        ' Preparation work
        Console.WriteLine("Running header and footer test sample…" & Environment.NewLine)

        SDKLicenseHelper.LicenseVerify()

        If Not Directory.Exists(outputPath) Then
            Directory.CreateDirectory(outputPath)
        End If
        If Not Directory.Exists(tempPath) Then
            Directory.CreateDirectory(tempPath)
        End If

        ' Sample 1: Export Annotation
        Dim annotationsDocument As CPDFDocument = CPDFDocument.InitWithFilePath("Annotations.pdf")

        If ExportAnnotation(annotationsDocument) Then
            Console.WriteLine("Export annotation done.")
        End If

        Console.WriteLine("--------------------")

        ' Sample 2: Import Annotations
        Dim document As CPDFDocument = CPDFDocument.InitWithFilePath("CommonFivePage.pdf")

        If ImportAnnotation(document) Then
            Console.WriteLine("Import annotation done.")
        End If

        Console.WriteLine("--------------------")

        Console.WriteLine("Done")
        Console.WriteLine("--------------------")
        Console.ReadLine()
    End Sub

    ' Export the annotations in the document to XFDF format
    Private Function ExportAnnotation(document As CPDFDocument) As Boolean
        Dim filePath As String = Path.Combine(outputPath, "ExportAnnotationTest.xfdf")
        If Not document.ExportAnnotationToXFDFPath(filePath, tempPath) Then
            Return False
        End If
        Console.WriteLine("Xfdf file in " & filePath)
        Return True
    End Function

    ' Importing XFDF into the document
    Private Function ImportAnnotation(document As CPDFDocument) As Boolean
        Dim filePath As String = Path.Combine(outputPath, "ImportAnnotationTest.pdf")

        If Not document.ImportAnnotationFromXFDFPath("Annotations.xfdf", tempPath) Then
            Return False
        End If
        If Not document.WriteToFilePath(filePath) Then
            Return False
        End If
        Console.WriteLine("Browse the changed file in " & filePath)
        Return True
    End Function

End Module

File: ComPDFKit for Windows/Examples/Samples/AnnotationImportExportTest/CS/AnnotationImportExportTest.cs
using ComPDFKit.PDFDocument;
using System;
using System.IO; 

namespace AnnotationImportExportTest
{
    internal class AnnotationImportExportTest
    {
        static private string parentPath = Path.GetDirectoryName(Path.GetDirectoryName(Path.GetDirectoryName(System.IO.Directory.GetCurrentDirectory())));
        static private string outputPath = Path.Combine(parentPath, "Output", "CS");
        private static string tempPath = Path.Combine(outputPath, "temp");

        static void Main(string[] args)
        {

            Console.WriteLine("Running header and footer test sample…" + Environment.NewLine);

            #region Preparation work

            SDKLicenseHelper.LicenseVerify();
             
            if (!Directory.Exists(outputPath))
            {
                Directory.CreateDirectory(outputPath);
            }
            if (!Directory.Exists(tempPath))
            {
                Directory.CreateDirectory(tempPath);
            }

            #endregion

            #region Sample 1: Export Annotation 

            CPDFDocument annotationsDocument = CPDFDocument.InitWithFilePath("Annotations.pdf"); 
            if (ExportAnnotaiton(annotationsDocument))
            {
                Console.WriteLine("Export annotaiton done.");
            }
            Console.WriteLine("--------------------");
            
            #endregion

            #region Sample 2: Import Annotations

            CPDFDocument document = CPDFDocument.InitWithFilePath("CommonFivePage.pdf"); 
            if (ImportAnnotaiton(document))
            {
                Console.WriteLine("Import annotaiton done.");
            }

            Console.WriteLine("--------------------");
            
            #endregion
             
            Console.WriteLine("Done");
            Console.WriteLine("--------------------");
            Console.ReadLine();
        }

        /// <summary>
        ///  Export the annotations in the document to XFDF format.
        ///  Parameters in function "ImportAnnotationFromXFDFPath":
        ///  1. path: The path to the exported XFDF file
        ///  2. tempPath: The path for storing temporary files. 
        /// </summary>
        /// <param name="document">A document with multiple annotations</param>
        /// <returns></returns>
        static private bool ExportAnnotaiton(CPDFDocument document)
        {
            var path = Path.Combine(outputPath, "ExportAnnotationTest.xfdf");
            if (!document.ExportAnnotationToXFDFPath(path, tempPath))
            {
                return false;
            }
            Console.WriteLine("Xfdf file in " + path);

            return true;
        }

        /// <summary>
        ///  Importing XFDF into the document.
        ///  Parameters in function "ImportAnnotationFromXFDFPath":
        ///  1. The path to the XFDF file for importing.
        ///  2. The path for storing temporary files.  
        /// </summary>
        /// <param name="document">A document without annotations used for importing XFDF.</param>
        /// <returns></returns>
        static private bool ImportAnnotaiton(CPDFDocument document)
        {
            var path = Path.Combine(outputPath, "ImportAnnotationTest.pdf");

            if (!document.ImportAnnotationFromXFDFPath("Annotations.xfdf", tempPath))
            {
                return false;
            }
            if (!document.WriteToFilePath(path))
            {
                return false;
            }
            Console.WriteLine("Browse the changed file in " + path); 
            return true;
        }
    }
}
